pipeline.py

```diff
@@ -27,62 +34,115 @@ class MorphologicalToken:
     """
     Stores language params for each processed token
     """
+
     def __init__(self, original_word, normalized_form):
-        pass
+        self.original_word = original_word
+        self.normalized_form = normalized_form
+        self.mystem_tags = ""
+        self.pymorphy_tags = ""
 
     def __str__(self):
-        return "MorphologicalToken instance here"
+        return f"{self.normalized_form}<{self.mystem_tags}>({self.pymorphy_tags})"
+
+    def sample(self):
+        raise NotImplementedError
 
 
 class CorpusManager:
     """
     Works with articles and stores them
     """
+
     def __init__(self, path_to_raw_txt_data: str):
-        pass
+        self.path_to_raw_txt_data = path_to_raw_txt_data
+        self._storage = {}
+
+        self._scan_dataset()
 
     def _scan_dataset(self):
         """
         Register each dataset entry
         """
-        pass
+        path = Path(self.path_to_raw_txt_data)
+
+        for idx in path.glob(r"*_raw.txt"):
+            article_id = re.match(r"^\d+", idx.parts[-1]).group()
+            self._storage[article_id] = Article(url=None, article_id=article_id)
 
     def get_articles(self):
         """
         Returns storage params
         """
-        pass
+        return self._storage
+
+    def sample(self):
+        raise NotImplementedError
 
 
 class TextProcessingPipeline:
     """
     Process articles from corpus manager
     """
+
     def __init__(self, corpus_manager: CorpusManager):
-        pass
+        self.corpus_manager = corpus_manager
+        self.text = ""
 
     def run(self):
         """
         Runs pipeline process scenario
         """
-        pass
+        for idx in self.corpus_manager.get_articles():
+            article = Article(url=None, article_id=idx)
+            self.text = article.get_raw_text()
+            processed_text = self._process()
+            article.save_processed(" ".join(map(str, processed_text)))
+
+    def sample(self):
+        raise NotImplementedError
 
     def _process(self) -> List[type(MorphologicalToken)]:
         """
         Performs processing of each text
         """
-        pass
+        result = Mystem().analyze(self.text)
+        morph = MorphAnalyzer()
+
+        processed_text: List[MorphologicalToken] = []
+
+        for token in result:
+
+            if token.get("analysis"):
+                morph_token = MorphologicalToken(
+                    token["text"], token["analysis"][0]["lex"]
```

Comment:
are you sure about other keys? why do you check only for `analysis` key existence? this part is very fragile

Reply:
addressed